openwisp_controller/geo/admin.py

```diff
@@ -107,6 +107,23 @@ def get_fields(self, request, obj=None):
             fields.remove("is_estimated")
         return fields
 
+    def change_view(self, request, object_id, form_url="", extra_context=None):
+        data = super().change_view(request, object_id, form_url, extra_context)
+        # Add after calling super() to ensure that latest object is fetched
+        # thus, avoiding duplicate messages.
+        obj = self.get_object(request, object_id)
+        if (
+            obj
+            and obj.is_estimated
+            and check_estimate_location_configured(obj.organization_id)
+        ):
+            messages.warning(
+                request,
+                "This location is estimated based on the device's last IP address."
+                " Please refine it for greater accuracy.",
```

Comment:
This needs to be flagged as translatable, but I am wondering, why don't we add this with a template?

Look for `class="warning"` in the codebase, it shouldn't be hard to do and it's going to be more reliable as it doesn't rely on cookies.